crates/nu-cmd-dataframe/src/dataframe/expressions/datepart.rs

```diff
@@ -0,0 +1,123 @@
+                SyntaxShape::String,
+                "Part of the date to capture.  Possible values are year, quarter, month, week, weekday, day, hour, minute, second, millisecond, microsecond, nanosecond",
+            )
+            .input_type(Type::Custom("expression".into()))
+            .output_type(Type::Custom("expression".into()))
+            .category(Category::Custom("expression".into()))
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        let example_value: DateTime<FixedOffset> = Utc::now().into();
+        vec![Example {
+            description: "Creates an expression to capture date parts",
+            example: "dfr col a | dfr datepart year | dfr into-nu",
+            result: {
+                let cols = vec!["expr".into(), "value".into()];
+                let expr = Value::test_string("column");
+                let value = Value::test_string("a");
+                let expr = Value::Record {
+                    cols,
+                    vals: vec![expr, value],
+                    span: Span::test_data(),
+                };
+
+                let cols = vec!["expr".into(), "datepart".into()];
+                let value = Value::test_date(example_value);
+
+                let record = Value::Record {
+                    cols,
+                    vals: vec![expr, value],
+                    span: Span::test_data(),
+                };
+
+                Some(record)
+            },
+        }]
+    }
+
+    fn search_terms(&self) -> Vec<&str> {
+        vec!["datepart", "date", "year", "month", "week", "weekday", "quarter", "day", "hour", "minute", "second", "millisecond", "microsecond", "nanosecond"]
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        let part: String = call.req(engine_state, stack, 0)?;
+
+        let expr = NuExpression::try_from_pipeline(input, call.head)?;
+        let expr_dt = expr.into_polars().dt();
+        let expr = match part.as_str() {
+            "year" => expr_dt.year(),
+            "quarter" => expr_dt.quarter(),
+            "month" => expr_dt.month(),
+            "week" => expr_dt.week(),
+            "day" => expr_dt.day(),
+            "hour" => expr_dt.hour(),
+            "minute" => expr_dt.minute(),
+            "second" => expr_dt.second(),
+            "millisecond" => expr_dt.millisecond(),
+            "microsecond" => expr_dt.microsecond(),
+            "nanosecond" => expr_dt.nanosecond(),
+            _ => {
+                return Err(ShellError::UnsupportedInput(
+                    format!("{} is not a valid datepart, expected one of year, month, day, hour, minute, second, millisecond, microsecond, nanosecond", part),
+                    "value originates from here".to_string(),
+                    call.head,
+                    call.head, // need to do better here
```

Comment:
need to do better? i guess that means you're not using unsupportedinput right?

Reply:
Sorry, here I think the span isn't right - I think it should be the span for the part of the datepart, but I wasn't super clear on how to traverse to it and wanted early directional feedback.

Reply:
I'd have to play a bit but I think if you get the `part` as a `Spanned<String>`, you'd have the span for the part.
```rust
let part: Spanned<String> = call.req(engine_state, stack, 0)?;
```
There's examples of how to get a spanned string in other commands i'm sure.